uk.co.compucorp.civicrm.hrleaveandabsences/CRM/HRLeaveAndAbsences/Mail/BaseRequestNotificationTemplate.php

```diff
@@ -0,0 +1,152 @@
+<?php
+
+use CRM_HRLeaveAndAbsences_BAO_LeaveRequest as LeaveRequest;
+use CRM_HRLeaveAndAbsences_Service_LeaveRequestComment as LeaveRequestCommentService;
+
+abstract class CRM_HRLeaveAndAbsences_Mail_BaseRequestNotificationTemplate {
+
+  /**
+   * @var \CRM_HRLeaveAndAbsences_Service_LeaveRequestComment
+   */
+  private $leaveRequestCommentService;
+
+  /**
+   * @var array|null
+   *   Stores the list of option values for the LeaveRequest status_id field.
+   */
+  private $leaveStatuses;
+
+  /**
+   * @var array|null
+   *   Stores the list of option values for the LeaveRequest from_date_type field.
+   */
+  private $leaveRequestDayTypes;
+
+  /**
+   * CRM_HRLeaveAndAbsences_Mail_BaseRequestNotificationTemplate constructor.
+   *
+   * @param \CRM_HRLeaveAndAbsences_Service_LeaveRequestComment $leaveRequestComment
+   */
+  public function __construct(LeaveRequestCommentService $leaveRequestComment) {
+    $this->leaveRequestCommentService = $leaveRequestComment;
+  }
+
+  /**
+   * Gets the message template for this Template Type.
+   *
+   * @return array
+   *  An array containing the message template attributes/values
+   */
+  abstract public function getTemplate();
```

Comment:
I still think it should return the ID, because that's what the actual code needs. On the tests, can't you load the Message Template by name (using the API) and then compare its ID with the value returned by the `getTemplateID()` method?